empyrical/stats.py

```diff
@@ -20,11 +20,102 @@
 from scipy import stats
 from six import iteritems
 
-from .utils import nanmean, nanstd, nanmin, up, down, roll
+from .utils import nanmean, nanstd, nanmin, up, down, roll, rolling_window
 from .periods import ANNUALIZATION_FACTORS, APPROX_BDAYS_PER_YEAR
 from .periods import DAILY, WEEKLY, MONTHLY, YEARLY
 
 
+def _create_unary_vectorized_roll_function(function):
+    def unary_vectorized_roll(arr, window, out=None, **kwargs):
+        """
+        Computes the {human_readable} measure over a rolling window.
+
+        Parameters
+        ----------
+        arr : array-like
+            The array to compute the rolling {human_readable} over.
+        window : int, required
+            Size of the rolling window in terms of the periodicity of the data.
+        out : array-like, optional
+            The array to store the store the output.
+        **kwargs
+            Forwarded to :func:`~empyrical.{name}`.
+
+        Returns
+        -------
+        rolling_{name} : array-like
+            The rolling {human_readable}.
+        """
+        allocated_output = out is None
+
+        if len(arr):
+            out = function(rolling_window(arr, window).T, out=out, **kwargs)
+        else:
+            out = np.empty(0, dtype='float64')
```

Comment:
I'm not sure what typical behavior is, but if the user passes `out`, do we need to zero the `out` array?

Reply:
This isn't zeroing the array, this is creating a length 0 array.